src/main/java/io/stargate/sgv2/jsonapi/exception/playing/APIException.java

```diff
@@ -0,0 +1,129 @@
+ *       instantiated at JVM start the error template is loaded.
+ *   <li>Create instances using the methods on {@link ErrorCode}.
+ * </ul>
+ *
+ * To get the Error to be returned in the {@link
+ * io.stargate.sgv2.jsonapi.api.model.command.CommandResult} call the {@link #get()} method to get a
+ * {@link CommandResponseError} that contains the subset of information we want to return.
+ */
+public abstract class APIException extends RuntimeException
+    implements Supplier<CommandResponseError> {
+
+  // All errors default to 200 HTTP status code, because we have partial failure modes.
+  // There are some overrides, e.g. a server timeout may be a 500, this is managed in the
+  // error config. See ErrorTemplate.
+  public static final int DEFAULT_HTTP_RESPONSE = Response.Status.OK.getStatusCode();
+
+  /**
+   * HTTP Response code for this error. NOTE: Not using enum from quarkus because do not want
+   * references to the HTTP framework this deep into the command processing
+   */
+  public final int httpResponse;
+
+  /** Unique identifier for this error instance. */
+  public final UUID errorId;
+
+  /** The family of the error. */
+  public final ErrorFamily family;
+
+  /**
+   * Optional scope of the error, inside the family.
+   *
+   * <p>Always none null, uses "" for no scope. See {@link ErrorScope}
+   */
+  public final String scope;
+
+  /** Unique code for this error, codes should be unique within the Family and Scope combination. */
+  public final String code;
+
+  /** Title of this exception, the same title is used for all instances of the error code. */
+  public final String title;
+
+  /**
+   * Message body for this instance of the error.
+   *
+   * <p>Messages may be unique for each instance of the error code, they are typically driven from
+   * the {@link ErrorTemplate}.
+   *
+   * <p>This is the processed body to be returned to the client. NOT called body to avoid confusion
+   * with getMessage() on the RuntimeException
+   */
+  public final String body;
+
+  public APIException(ErrorInstance errorInstance) {
+    Objects.requireNonNull(errorInstance, "ErrorInstance cannot be null");
+
+    this.errorId = errorInstance.errorId();
+    this.family = errorInstance.family();
+    this.scope = errorInstance.scope().scope();
+    this.code = errorInstance.code();
+    this.title = errorInstance.title();
+    this.body = errorInstance.body();
+    Objects.requireNonNull(
+        errorInstance.httpResponseOverride(), "httpResponseOverride cannot be null");
+    this.httpResponse = errorInstance.httpResponseOverride().orElse(DEFAULT_HTTP_RESPONSE);
+  }
+
+  public APIException(
+      ErrorFamily family, ErrorScope scope, String code, String title, String body) {
+    this(new ErrorInstance(UUID.randomUUID(), family, scope, code, title, body, Optional.empty()));
+  }
+
+  @Override
+  public CommandResponseError get() {
+    return null;
```

Comment:
@amorton Why isn't this left abstract to require sub-classes to implement?

Reply:
it is removed in later changes, was a POC placeholder